activerecord/lib/active_record/query_logs.rb

```diff
@@ -6,7 +6,31 @@
 module ActiveRecord
   # = Active Record Query Logs
   #
-  # Automatically tag SQL queries with runtime information.
+  # Automatically append comments to SQL queries with runtime information tags. This can be used to trace troublesome
+  # statements found using database tooling back to the area of the application code where these statements were
+  # generated by Active Record.
+  #
+  # Query logs can be enabled via Rails configuration in <tt>config/application.rb</tt> or any environment initializer:
+  #
+  #     config.active_record.query_log_tags_enabled = true
+  #
+  # or directly:
+  #
+  #     ActiveRecord.query_log_tags_enabled = true
+  #
+  # By default the name of the application, the name and action of the controller, or the name of the job are logged
+  # using the {SQLCommenter}[https://open-telemetry.github.io/opentelemetry-sqlcommenter/] format.
+  # The tags shown in a query comment can be configured via Rails configuration, wrapped in an +after_initialize+ block:
+  #
+  #     config.after_initialize do
```

Comment:
Why does it need to be `after_initialize`? `config.active_record.query_log_tags_enabled = true` was not, and so the inconsistency stands out.

Reply:
See the second snippet of https://github.com/rails/rails/issues/46103 and @skipkayhil explanation of load order. Without the after initialize block the customized tags would be appended with the default tags, resulting in duplicates in the output.

This approach worked in my test app `application.rb` and IMO reads less confusing than explaining which configuration needs to go in which files, but I agree it looks a little odd still and might not be the best way. Definitely leveraging some Cunningham's Law here 😁

Reply:
Yeah, recommending `after_initialize` is kind of unfortunate. The alternatives would be an `initializer` in `config/application` or `config/initializers/`, but as mentioned above it has to be in a different place from `query_log_tags_enabled` if you want full control over the list of tags.

Reply:
Hmmm, I think something else is going on in that issue.  Check out this test: https://github.com/rails/rails/blob/0eb42cae30e3b74d88771c971d2d2957084cf88a/railties/test/application/query_logs_test.rb#L95..L102 - that is not using `after_initialize`.

And there's no duplicates either... the comment I got when I ran the test was `/*pid='17125',controller='users',action='index'*/`.

Reply:
Ohhh I see what's going on there. I think that is a bug: https://github.com/rails/rails/pull/46279

Reply:
Thanks. Assuming that gets merged, I removed the after_initialize suggestions here.